src/goto-programs/goto_convert.cpp

```diff
@@ -1142,7 +1142,11 @@ void goto_convertt::convert_assert(
   t->guard.swap(cond);
   t->source_location=code.source_location();
   t->source_location.set(ID_property, ID_assertion);
-  t->source_location.set("user-provided", true);
+  irep_idt property_class=t->source_location.get_property_class();
+  // if it's not our instrumentation for runtime exceptions, then it must be
+  // provided by the user
+  if(id2string(property_class)!="runtime-exception")
```

Comment:
If at all, I suggest to do that in a separate PR since it involves replacing all property class names in goto_check. This could fix the inconsistent naming using spaces vs dashes. However, an important side issue is that the property class is printed 'as is' in the tool output. This can cause problems with tools that integrate CBMC and rely on these names. Also our SV-COMP wrapper scripts use these names.